### PR TITLE
Week 3 - Part 2

### PR DESCRIPTION
**Pull request by tanghengyu**
**ConvNet Evolutions, Architectures, Implementation Details and Advantages**

We give an introduction on ConvNet evolutions. We discuss in detail on architectures of ConvNet with PyTorch implementation of LeNet5, exemplified by the task of digit recognition on MNIST. Based on its design principles, we expand on the advantages of ConvNet which fully exploit compositionality, stationary, and locality features of natural images.

### CODE REVIEW

**Comment by Atcold** [comment]
Edit the first message, here above, and add the segment title and 50 word summary.

**Comment by tanghengyu** [comment]
First message modified to include segment titles and summary.

File: docs/chapters/03-2.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+Generic CNN architecture can broken down into several basic layer archetypes:
+
+* Normalization layers - adjusting whitening (optional)
+  * Subtratrive methods e.g. average removal, high pass filtering
+  * Divisive: local contrast normalization, variance normalization
+* Filter Banks: increasing dimensionality, finding edges, etc.
+* Non-linearities: sparsification
+  * Typically Rectified Linear Unit (ReLU): $\text{ReLU}(x) = \text{max}(x, 0)$
+* Pooling: aggregating over a feature map:
+  * $\text{MAX}= \text{Max}_i(X_i)$
```
Comment (Atcold) [inline docs/chapters/03-2.md]: What is this???

File: docs/chapters/03-2.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  * $\text{L}p= \left(\sum_{i=1}^n \|X_i\|^p \right)^{\frac{1}{p}}$
+  * $\text{Prob}= \frac{1}{b} \left(\sum_{i=1}^n e^{b X_i} \right)$
```
Comment (Atcold) [inline docs/chapters/03-2.md]: Ugly!

**Comment by Atcold** [comment]
There is no `03-2/detailed_convNet.png`.

**Comment by Atcold** [comment]
Change all `mxn` into `$m \times n$`, thanks.

File: docs/chapters/03-2.md (hunk 2 above)
Comment (marikgoldstein) [reply docs/chapters/03-2.md]: Alf, what is ugly? Do you mean you prefer \texttt{MAX} instead of \text{MAX} or something?

**Pull request merged**